### ISSUE TITLE
Evaluate a mask-penalty auxiliary training term

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#97]
## Goal

Evaluate whether adding a mask-penalty term to the training objective reduces
probability mass on illegal moves enough to justify its cost, and remove it if
it does not.

## Background

`docs/evaluation.md` already contemplates this:

```text
loss = move_ce + lambda * mask_penalty
```

and deliberately gates it, because human move cross-entropy already pushes
probability toward legal moves -- the target move is always legal. The gate is
that evaluation should first show the model continues to place too much
probability on illegal moves.

That gate cannot be judged against the current baseline, which is a small model
stopped by its step bound rather than by convergence, with raw-logit legality
still improving when the run ended. Measured mask penalty fell from 1.63 at step
1,000 to 0.75 at step 8,000 with no plateau. This work should therefore start
from a checkpoint that has already received the ordinary improvements in this
milestone, so the measurement describes a converged model rather than an
undertrained one.

## Scope

A training-objective experiment with an explicit removal condition. Not a
runtime change: legal masking remains the guarantee that illegal moves are
never played, and this only affects what the raw distribution learns.

## Acceptance Criteria

- The auxiliary term is opt-in through configuration with an explicit weight,
  and runs record whether it was active and at what weight.
- A baseline without the term and a run with it are compared on the same data,
  model, and budget, using the frozen pool.
- Comparison covers raw-logit legality, held-out move prediction, and the
  training and inference cost of the term, so a legality gain that costs move
  quality or throughput is visible rather than hidden.
- Legality results are reported per game phase and per rule case, since phase
  dominates the absolute level and the term is expected to matter most where
  legality is worst.
- Cost is measured through the benchmarks in #60 rather than a bespoke timing
  path.
- A stated removal condition is recorded with the result: if the term does not
  improve legality by a margin worth its cost, it is removed rather than left
  enabled by default.
- Fixture tests cover the loss composition, the disabled path producing the
  prior objective exactly, and deterministic behavior.

## Notes

The honest null result here is that human cross-entropy on a converged model
already handles legality, and the term buys nothing. That outcome should be
recorded and the term deleted, not left behind a default-off flag that later
work has to reason about.

Weight selection is the obvious failure mode: a weight large enough to move
legality may distort move preferences among legal moves, which is the thing the
project most wants to keep faithful to human play. Whatever weight is chosen,
the comparison must show move-prediction quality did not pay for the legality
gain.

Several weights read against one control arm, as a dial rather than as
independent candidates: state the response expected across the weights before
the arms run, and locate where the guard metric turns. `docs/evaluation.md`
(Regression Comparisons) owns why that is one claim rather than several.

## Carrying Metric

**Shape: several named metrics.** The acceptance criteria already state the
conjunction: legality improves, move prediction does not pay for it, and the
cost is bounded. Partial movement is the removal condition firing rather than a
pass.

| metric | benchmark | floor today |
| --- | --- | --- |
| `legality.mask_penalty`, and the per-phase and per-rule-case `legality.mask_penalty_*` | `anthro eval run` | prints one |
| `novelty.mask_penalty`, `novelty.legal_mass` | `anthro eval novelty` | prints one |
| `held_out.move_loss` — the guard, must not degrade | `anthro eval run` | prints one |
| `inference.forward_throughput_per_second` — the inference cost | `anthro eval inference` | prints one |
| `training.active_positions_per_second` — the training cost | `anthro train` | declares none |

`legality` is the family where every metric, all 16, carries a dispersion on its
reading, so `anthro eval report` prints a combined floor for each and the
removal condition can be stated in advance as a margin against it rather than as
a judgement made after the reading. `docs/evaluation.md` (Regression
Comparisons) sets that margin's starting value.

The two floored `novelty` metrics are exactly the pair this term targets — the
benchmark bootstraps its own data-sampling floor over each arm's derived games
for those two and no others — so `anthro eval novelty` is a second instrument
for the same claim rather than an unrelated family.

The two cost legs are qualified differently, since
`docs/decisions/0043-a-delta-floor-is-combined-from-the-two-readings-it-compares.md`
collapsed the floor system to one dispersion per reading. `anthro eval
inference` replicates itself across processes and reports the spread beside its
value, so an inference-cost delta meets a floor like any other. Every
`training.*` metric instead declares `no_sampling_floor_reason` under
`docs/decisions/0061-a-training-cost-reading-has-no-replicate-to-resample.md`
and renders `unqualifiable`, because a replicate of a training reading is a
second training run. A training-cost claim therefore rests on the arms
themselves rather than on a verdict.

**Comment by mmmfrieddough** [comment]
## Closing: the gate is not met, by three orders of magnitude

This issue gates itself on evaluation first showing that the model continues to
place too much probability on illegal moves. Read on the ablation vehicle
(`arm-488-seed61`, step 69465, swept 2026-09-02), it does not.

| metric | when this was filed | vehicle |
| --- | --- | --- |
| `legality.mask_penalty` | 1.63 at step 1,000, 0.75 at step 8,000 | **0.000567** |
| `legality.legal_mass` | | 0.99945 |
| `legality.top1_illegal_rate` | | 0.000056 |

Worst rule cases and every phase:

```
promotion      0.00347        opening      0.00030
en_passant     0.00196        middlegame   0.00077
pin            0.00124        endgame      0.00045
only_move      0.00105
```

Two things follow beyond the headline. The phase gradient this issue predicted
has inverted: it expected legality to be worst in the endgame, where the earlier
reading had 1.26 against 0.19 in the opening, and the opening is now the cleanest
slice with all three negligible. And the off-distribution case does not rescue
it either: `novelty.mask_penalty_delta` rises about tenfold across the dose
range, from 0.00016 to 0.00162, which is a tenfold increase on a quantity that
starts at almost nothing.

So the removal condition can be evaluated without running the arms. Available
headroom is roughly 0.0006 nats. Against that sits the risk this issue names
itself: a weight large enough to move legality distorts preference among legal
moves, which is the property the project most wants to keep faithful. That is an
asymmetric trade against a gain that rounds to zero.

Closing as not planned, with the readings above as the recorded gate result.
That is the outcome the Notes section asked for: the honest null is that human
cross entropy on a converged model already handles legality, and the term is
deleted rather than left behind a default-off flag that later work has to reason
about.
